src/architecture.jl

```diff
@@ -96,15 +96,15 @@ A device that represents a serial CPU execution.
 """
 struct Serial <: AbstractTask end
 serial() = Serial()
-get_device(::Serial) = CPU(1)
+get_device(::Serial, id=1) = CPU(id, 1)
```

Comment:
can we default the id to a symbol like `:default` to avoid using integers, which have confusion with the number of cores (which we probably don't want here)?